astronomer/providers/snowflake/sensors/snowflake.py

```diff
@@ -74,30 +75,61 @@ def __init__(
     def execute(self, context: Context) -> None:
         """Check for query result in Snowflake by deferring using the trigger"""
         if not poke(self, context):
-            self.defer(
-                timeout=timedelta(seconds=self.timeout),
-                trigger=SnowflakeSensorTrigger(
-                    sql=self.sql,
-                    poke_interval=self.poke_interval,
-                    parameters=self.parameters,
-                    success=self.success,
-                    failure=self.failure,
-                    fail_on_empty=self.fail_on_empty,
-                    dag_id=context["dag"].dag_id,
-                    task_id=context["task"].task_id,
-                    run_id=context["dag_run"].run_id,
-                    snowflake_conn_id=self.snowflake_conn_id,
-                ),
-                method_name=self.execute_complete.__name__,
-            )
+            self._defer(context)
 
-    def execute_complete(self, context: Context, event: dict[str, str] | None = None) -> Any:
+    def _validate_result(self, result: list[tuple[Any]]) -> Any:
+        """Validates query result and verifies if it returns a row"""
+        if not result:
+            if self.fail_on_empty:
+                raise AirflowException("No rows returned, raising as per fail_on_empty flag")
+            else:
+                return False
+
+        first_cell = result[0][0]
+        if self.failure is not None:
+            if callable(self.failure):
+                if self.failure(first_cell):
+                    raise AirflowException(f"Failure criteria met. self.failure({first_cell}) returned True")
+            else:
+                raise AirflowException(f"self.failure is present, but not callable -> {self.failure}")
+        if self.success is not None:
+            if callable(self.success):
+                return self.success(first_cell)
+            else:
+                raise AirflowException(f"self.success is present, but not callable -> {self.success}")
+        return bool(first_cell)
+
+    def _defer(self, context: Context) -> None:
```

Comment:
why is this separate method required

Reply:
I think it's a separate method used in execute_complete

Reply:
it has been used in execute_complete and execute so to avoid duplicate code